### DISCUSSION TITLE
Safari 15.4 broke x-transition.opacity

### DISCUSSION THREAD

**Discussion by miagg** [alpinejs/alpine#2761] [7. Bugs (archived)]
I have a strange issue when using the latest Safari version (15.4) under Mac OS Monterey 12.3.

If I use `x-transition.opacity.duration.300ms` or `x-transition.opacity` in general I get a flash frame after transition ends.

It works absolutely fine on every other browser, and I am 100% sure it also worked on Safari before I update.

Any ideas on what is causing the issue?

**Comment by magnusskare** [comment]
I completely missed this, and just made a similar report here, also running: #2827 

Also running Safari 15.4 in the example. Does this look similar to the issues you were experiencing?

**Comment by miagg** [comment]
UPDATE: To make it even more interesting, I discovered that if I recreate the transition manually it solves the issue.

for example if instead of this:
```
x-transition.opacity.duration.300ms
```

I use this:
```
x-transition:enter="transition ease-out duration-300"
x-transition:enter-start="opacity-0"
x-transition:enter-end="opacity-100"
x-transition:leave="transition ease-in duration-300"
x-transition:leave-start="opacity-100"
x-transition:leave-end="opacity-0"
```

...then everything is back to normal ¯\\_(ツ)\_/¯

Maybe some kind of race condition? It flashes for exactly one frame.
Every single alpinejs site using x-transition is broken on Mac OS Monterey 12.3.

You can illustrate this by going to [https://laravel.com](https://laravel.com) and toggling the help menu from Safari 15.3 or iOS 15.4 device
